### PR TITLE
Fix context propagation in Filters

### LINKED ISSUE #9719: Propagated context happening before security filter phase is cleared when Micronaut Security is enabled
### Expected Behavior

I do not know if this is intended behaviour or a bug. I believe it is a bug and therefore I created it as a bug type. 

We use [context propagation ](https://docs.micronaut.io/latest/guide/#contextPropagation) to propagate some custom tracing information. However when we enable Micronaut Security the context is removed if it is applied before the Micronaut Security Filter phase. Is this intended to prevent some kind of attack or is this a bug? 
I would expect the tracing context to remain around the security concerns. 

### Actual Behaviour

Enabling Micronaut Security remove the propagated context that was added before. 

### Steps To Reproduce

See linked repository for a failing test but in short for some immediate context
```
@Filter(MATCH_ALL_PATTERN)
class CustomTracingFilter : HttpServerFilter {

    override fun getOrder(): Int {
        return ServerFilterPhase.TRACING.after()
        // Replace order to after security to make the test pass
//        return ServerFilterPhase.SECURITY.after()
    }

    override fun doFilter(request: HttpRequest<*>, chain: ServerFilterChain): Publisher<MutableHttpResponse<*>> {
        val trace = request.headers.get("X-Trace")
        PropagatedContext.get().plus(MyPropagationContext(trace)).propagate().use {
            return chain.proceed(request)
        }
    }
}
```
Either by placing the filter after SECURITY or by setting `micronaut.security.enabled=false` will "fix" the test. 

### Environment Information

_No response_

### Example Application

https://github.com/olamelin/micronaut-security-tracing

### Version

4.0.3

### PR DESCRIPTION
**Pull request by dstepanov**
Fixes https://github.com/micronaut-projects/micronaut-core/issues/9719

### CODE REVIEW

**Comment by sdelamo** [comment]
@dstepanov ci failed. Can you check it?

File: http/src/main/java/io/micronaut/http/filter/FilterRunner.java [hunk 1]
```diff
@@ -552,8 +551,8 @@ private static FilterReturnHandler prepareReturnHandler(ConversionService conver
                     return next.handle(context, null, continuation);
                 }
 
-                Mono<?> publisher = Mono.from(Publishers.convertPublisher(conversionService, returnValue, Publisher.class));
-
+                Mono<?> monoPublisher = Mono.from(Publishers.convertPublisher(conversionService, returnValue, Publisher.class));
+                Publisher<?> publisher = ReactivePropagation.propagate(context.propagatedContext, monoPublisher);
```
Comment (yawkat) [inline http/src/main/java/io/micronaut/http/filter/FilterRunner.java]: I think the Mono.from should happen after the propagate call

**Review by yawkat — APPROVED** [review]

File: http/src/main/java/io/micronaut/http/filter/FilterRunner.java (hunk 1 above)
Comment (sdelamo) [reply http/src/main/java/io/micronaut/http/filter/FilterRunner.java]: @dstepanov can you address this?

Comment (dstepanov) [reply http/src/main/java/io/micronaut/http/filter/FilterRunner.java]: @yawkat I don't think we need that mono call anyway

Comment (yawkat) [reply http/src/main/java/io/micronaut/http/filter/FilterRunner.java]: yea im not sure. ReactivePropagation.propagate has some logic that branches on Flux vs Mono, which after this PR might hit the different branch. But I doubt there's any impact.

**Review by sdelamo — APPROVED** [review]

**Pull request merged**